tests/jballerina-unit-test/src/test/resources/test-src/object/object-referencing-module-variable.bal

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+final int classI = 111222;
+
+class ModuleVariableReferencingClass {
+    int i = classI;
+}
+
+function value(int k = classI) returns int {
+    return k;
+}
+
+ModuleVariableReferencingClass c1 = new;
+
+function testClassReferencingModuleVariable() {
+    ModuleVariableReferencingClass c = new;
+    assertEquality(111222, c.i);
+    assertEquality(111222, c1.i);
+}
+
+function testLocalObjectConstructorReferencingModuleVariable() {
+    var value = object {
+        int i = classI;
+    };
+    assertEquality(111222, value.i);
+}
+
+function assertEquality(any|error expected, any|error actual) {
```

Comment:
Aren't you only passing `anydata` atm? Please update the function accordingly.

Reply:
Yes. Updated.